src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -332,15 +332,62 @@ internal async Task<Dictionary<string, string>> GetProcessEnvironmentAsync(Cance
             return await helper.ReadEnvironmentAsync(response.Continuation, token).ConfigureAwait(false);
         }
 
-        internal void ApplyStartupHook(string startupHookPath)
+        /// <summary>
+        /// Loads the specified assembly with a StartupHook in the target process.
+        /// </summary>
+        /// <param name="startupHookPath">The path to the assembly containing the StartupHook.</param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="startupHookPath"/> is null or empty.</exception>
+        /// <exception cref="FileNotFoundException">Thrown when the specified assembly does not exist.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the runtime version is less than 8.0.</exception>
+        public void ApplyStartupHook(string startupHookPath)
         {
+            if (string.IsNullOrEmpty(startupHookPath))
+            {
+                throw new ArgumentNullException(nameof(startupHookPath));
+            }
+
+            if (!File.Exists(startupHookPath))
+            {
+                throw new FileNotFoundException($"Startup hook file not found: {startupHookPath}");
+            }
+
+            ProcessInfo processInfo = GetProcessInfo();
+            if (!processInfo.TryGetProcessClrVersion(out Version version) || version.Major < 8)
```

Comment:
This check should be unnecessary since down-version runtimes will respond that they do not understand the command and the ValidateResponseMessage call will throw an appropriate exception.

Reply:
Fixed via bb7c6cb